src/sentry/plugins/providers/integration_repository.py

```diff
@@ -122,6 +151,13 @@ def get_repository_data(self, organization, config):
         """
         return config
 
+    def update_repository_data(self, repo, new_data):
+        """
+        Update repo in database if necessary. Return True if update happened,
+        False otherwise.
+        """
+        raise NotImplementedError
```

Comment:
I'm leaving this up to to individual integrations to implement since each gets its data from slightly different sources (different attribute names in the API response) and since each has its own `repo.config` structure.

Reply:
Indeed, that (implementing it for all the integrations) was my plan, but wanted to get generalized approval of the idea/method before I did it for all of them. Once things look good with the bitbucket one, I can do the rest.

Reply:
To me, this is a nice to have feature rather than something I think the other integrations should be forced to have. But I won't push back hard on this one as long as we're avoiding the notimplemented error

Reply:
My personal opinion is that this is an edge case and more of a nice to have than something I'd inherently feel was required by the integrations. But I think if you're wanting to implement and write tests for it that's great. :) I would just urge you to turn it into its separate pr.